sdk/keyvault/remove-test-resources-pre.ps1

```diff
@@ -0,0 +1,79 @@
+
+#Requires -Version 6.0
+#Requires -PSEdition Core
+
+# Use same parameter names as declared in eng/common/TestResources/Remove-TestResources.ps1 (assume validation therein).
+[CmdletBinding(DefaultParameterSetName = 'Default', SupportsShouldProcess = $true, ConfirmImpact = 'Medium')]
+param (
+    [Parameter()]
+    [string] $ResourceGroupName,
+
+    [Parameter()]
+    [string] $TenantId,
+
+    [Parameter()]
+    [string] $ProvisionerApplicationId,
+
+    [Parameter()]
+    [string] $ProvisionerApplicationSecret,
+
+    [Parameter()]
+    [string] $SubscriptionId,
+
+    # Captures any arguments from eng/New-TestResources.ps1 not declared here (no parameter errors).
+    [Parameter(ValueFromRemainingArguments = $true)]
+    $RemainingArguments
+)
+
+# By default stop for any error.
+if (!$PSBoundParameters.ContainsKey('ErrorAction')) {
+    $ErrorActionPreference = 'Stop'
+}
+
+function Log($Message) {
+    Write-Host ('{0} - {1}' -f [DateTime]::Now.ToLongTimeString(), $Message)
+}
+
+function PurgeKeyVault($Vault) {
+  Log "Deleting Key Vault named '$($Vault.VaultName)'"
+  Remove-AzKeyVault -Name "$($Vault.VaultName)" -ResourceGroupName "$($Vault.ResourceGroupName)" -Location $($Vault.Location) -Force
+  Log "Deleted."
+
+  Log "Purging Key Vault named '$($Vault.VaultName)'"
+  Remove-AzKeyVault -Name "$($Vault.VaultName)" -Location $($Vault.Location) -InRemovedState -Force
+
+  Log "'$($Vault.VaultName)' successfully deleted and purged."
+}
+
+function PurgeManagedHsm($ManagedHsm) {
+  Log "Deleting Managed HSM named '$($ManagedHsm.Name)'"
+  az keyvault delete --resource-group "$ResourceGroupName" --hsm-name "$($ManagedHsm.Name)"
+  Log "Deleted."
+
+  Log "Purging Managed HSM named '$($ManagedHsm.Name)'"
+  az keyvault purge --hsm-name "$($ManagedHsm.Name)"
+
+  Log "$($ManagedHsm.Name) successfully deleted and purged."
+}
+
+Log "Permanently deleting all Key Vaults in resource group $ResourceGroupName"
+Get-AzKeyVault -ResourceGroupName $ResourceGroupName | ForEach-Object { PurgeKeyVault($_) }
+
+# TODO: Use Az module when available; for now, assumes Azure CLI is installed and in $Env:PATH.
+if ($ProvisionerApplicationId -and $ProvisionerApplicationSecret -and $TenantId) {
+  Log "Logging '$ProvisionerApplicationId' into the Azure CLI"
+  az login --service-principal --tenant "$tenantId" --username "$ProvisionerApplicationId" --password="$ProvisionerApplicationSecret" --output none
+  Log "Setting the subscription for the logged in user"
+  az account set --subscription "$SubscriptionId"
+} else {
+  Log "No credentials provided; skipping Azure CLI login and assuming current user is logged in and is using the correct subscription."
+}
+
+Log "Permanently deleting all Managed HSMs in resource group $ResourceGroupName"
+az keyvault list --resource-type hsm --resource-group "$ResourceGroupName" | ConvertFrom-Json | ForEach-Object { PurgeManagedHsm($_)}
+
+Log "Successfully deleted and purged all Key Vaults and Managed HSMs."
```

Comment:
I recommend logging out of the CLI so the user doesn't get confused later. Unlike the PowerShell cmdlets, the CLI doesn't appear to have a way to scope the login to the current process. So around here, run `az logout --username "$ProvisionerApplicationId"`